### PR TITLE
(Re)populate ROS_PACKAGE_PATH from all underlays

### LINKED ISSUE #624: rosenv does not correctly consider CMAKE_PREFIX_PATH in AFTER_SCRIPT
I made the following observation:
- using Github Actions (does NOT happen on travis)
- for `noetic` :point_right: default builder is `colcon` (does NOT happen on `kinetic` :point_right: default builder `catkin-tools`)
- using `UNDERLAY` option with custom docker image
- `AFTER_SCRIPT: 'rosenv bash .travis.xacro_test.sh'` 
  (see https://github.com/ipa320/cob_common/blob/kinetic_dev/.travis.yml#L12)

for some reason in GHA noetic job, the test fails because the resources for the xacro test cannot be found, although they should be findable in the `$CMAKE_PREFIX_PATH`

```
   Starting function 'after_script'
  Testing /home/runner/work/mojin_common/mojin_common/mojin_description/urdf/base_theta/base_theta.urdf.xacro
  Testing /home/runner/work/mojin_common/mojin_common/mojin_description/urdf/mrl_base/base.gazebo.xacro
  Testing /home/runner/work/mojin_common/mojin_common/mojin_description/urdf/mrl_base/base.urdf.xacro
  resource not found: cob_description
  ROS path [0]=/opt/ros/noetic/share/ros
  ROS path [1]=/root/target_ws/install/mojin_common/share
  ROS path [2]=/root/target_ws/install/mojin_utils/share
  ROS path [3]=/root/target_ws/install/mojin_msgs/share
  ROS path [4]=/root/target_ws/install/mojin_description/share
  ROS path [5]=/opt/ros/noetic/share
```

the missing resource `cob_description` is part of the `robot_ws` - see `CMAKE_PREFIX_PATH` below

I'm not sure whether my `AFTER_SCRIPT` is not colcon-ready or whether `rosenv` is doing something incorrect

EDIT:
here is the output of `AFTER_SCRIPT: 'rosenv && echo "$CMAKE_PREFIX_PATH" && echo "$UNDERLAY"'`
- for kinetic:
   ```
  Starting function 'after_script'
  /root/target_ws/install:/root/git/app_ws/install:/root/git/rosjava_ws/install:/root/git/mojin_ws/install:/root/git/nav_ws/install:/root/git/robot_ws/install:/opt/ros/kinetic:/root/git/care-o-bot/devel
  /root/git/care-o-bot/devel
   <<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
   Function 'after_script' returned with code '0' after 0 min 0 sec
   ```
- for noetic:
   ```
  Starting function 'after_script'
  /root/target_ws/install/mojin_utils:/root/target_ws/install/mojin_msgs:/root/target_ws/install/mojin_description:/root/git/app_ws/install:/root/git/rosjava_ws/install:/root/git/mojin_ws/install:/root/git/nav_ws/install:/root/git/robot_ws/install:/opt/ros/noetic
  /root/git/care-o-bot/devel
   <<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
   Function 'after_script' returned with code '0' after 0 min 0 sec
   ```

### PR DESCRIPTION
**Pull request by ipa-mdl**
Fixes #624

@fmessmer: please test

### CODE REVIEW

File: industrial_ci/src/workspace.sh [hunk 1]
```diff
@@ -333,3 +333,19 @@ function ici_test_workspace {
     ici_run "run_${name}_test" builder_run_tests "$extend" "$ws"
     builder_test_results "$extend" "$ws"
 }
+
+function ici_source_setup {
+  echo ici_with_unset_variables source "$1/setup.bash"
```
Comment (ipa-mdl) [inline industrial_ci/src/workspace.sh:338]: Oops
```suggestion
ici_with_unset_variables source "$1/setup.bash"
```

**Comment by fmessmer** [comment]
thanks!
I will try to find some time to test this...
...might not be before next week

**Comment by fmessmer** [comment]
did it quicker than I thought :wink: 
`AFTER_SCRIPT: 'rosenv && echo "$CMAKE_PREFIX_PATH" && echo "$UNDERLAY" && echo "$ROS_PACKAGE_PATH"'`
- for noetic with not having set `BUILDER`, i.e. using `colcon`
```
  Starting function 'after_script'
  /root/target_ws/install/mojin_utils:/root/target_ws/install/mojin_msgs:/root/target_ws/install/mojin_description:/root/git/app_ws/install:/root/git/rosjava_ws/install:/root/git/mojin_ws/install:/root/git/nav_ws/install:/root/git/robot_ws/install:/opt/ros/noetic
  /root/git/care-o-bot/devel
  /root/target_ws/install/mojin_utils/share:/root/target_ws/install/mojin_msgs/share:/root/target_ws/install/mojin_description/share:/root/git/app_ws/install/share:/root/git/rosjava_ws/install/share:/root/git/mojin_ws/install/share:/root/git/nav_ws/install/share:/root/git/robot_ws/install/share:/opt/ros/noetic/share
<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
Function 'after_script' returned with code '0' after 0 min 0 sec
```
- for noetic with `BUILDER: catkin_tools`
```
   Starting function 'after_script'
  /root/target_ws/install:/root/git/app_ws/install:/root/git/rosjava_ws/install:/root/git/mojin_ws/install:/root/git/nav_ws/install:/root/git/robot_ws/install:/opt/ros/noetic:/root/git/care-o-bot/devel
  /root/git/care-o-bot/devel
  /root/target_ws/install/share:/root/git/app_ws/install/share:/root/git/rosjava_ws/install/share:/root/git/mojin_ws/install/share:/root/git/nav_ws/install/share:/root/git/robot_ws/install/share:/opt/ros/noetic/share
<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
Function 'after_script' returned with code '0' after 0 min 0 sec
```

comparing this to https://github.com/ros-industrial/industrial_ci/issues/624#issuecomment-779241521, it seems that the `ROS_PACKAGE_PATH` for `BUILDER=colcon` not correctly includes the `UNDERLAY`

the test reported to fail in #624 pass with this feature branch and `colcon` as builder for noetic :+1:

**Comment by ipa-mdl** [comment]
>  it seems that the ROS_PACKAGE_PATH for BUILDER=colcon not correctly includes the UNDERLAY

typo?

**Comment by fmessmer** [comment]
> > it seems that the ROS_PACKAGE_PATH for BUILDER=colcon not correctly includes the UNDERLAY
> 
> typo?

yes, sorry

**Pull request merged**